projects/bp-gallery/src/components/views/geoguessr/GeoView.tsx

```diff
@@ -0,0 +1,154 @@
+import './GeoView.scss';
+
+const getAllPictureIds = (archives: FlatArchiveTag[]) => {
+  const list: string[] = [];
+  archives.forEach(archive => archive.pictures?.forEach(picture => list.push(picture.id)));
+  return list;
+};
+
+const shufflePictureIds = (pictureIds: string[], seed: number) => {
+  const newPictureIds = [...pictureIds];
+  for (let i = newPictureIds.length - 1; i > 0; i--) {
+    const j = Math.abs(Math.floor(i * Math.sin(i % seed)));
+    [newPictureIds[i], newPictureIds[j]] = [newPictureIds[j], newPictureIds[i]];
+  }
+  return newPictureIds;
+};
+
+const getTodaysPictureQueue = (pictureIds: string[]) => {
+  const pictureNumber = 10;
+  const currentDate = new Date();
+  const startDate = new Date(currentDate.getFullYear(), 0, 1);
+  const days = Math.floor((currentDate.valueOf() - startDate.valueOf()) / (24 * 60 * 60 * 1000));
+  const resultIndex = pictureIds.length % days;
+  const list = [];
+  for (let i = 0; i < pictureNumber; i++) {
+    list.push(pictureIds[(resultIndex + i) % pictureIds.length]);
+  }
+  return list;
+};
+
+const GeoView = () => {
+  const { t } = useTranslation();
+  const fallbackPictureId = '3';
+  const pictureQueue = useRef(['']);
+  const seed = 42;
+
+  const hasReadInstructions = Boolean(
+    JSON.parse(localStorage.getItem('hasReadInstructions') ?? 'false')
+  );
+  const [modalOpen, setModalOpen] = useState(!hasReadInstructions);
+  const dontShowAgain = () => {
+    localStorage.setItem('hasReadInstructions', 'true');
+    setModalOpen(false);
+  };
+
+  const { data: picturesData } = useGetAllPicturesByArchiveQuery();
+  const archives: FlatArchiveTag[] | undefined =
+    useSimplifiedQueryResponseData(picturesData)?.archiveTags;
+
+  useEffect(() => {
+    const allPictureIds = archives ? getAllPictureIds(archives) : [];
+    const shuffledPictureIds = shufflePictureIds(allPictureIds, seed);
+    pictureQueue.current = getTodaysPictureQueue(shuffledPictureIds);
+    setPictureId(getNextPicture);
+  }, [archives]);
+
+  const getNextPicture = () => {
+    const nextPicture = pictureQueue.current[0];
+    pictureQueue.current = pictureQueue.current.filter((elem, index) => index !== 0);
+    return nextPicture;
+  };
+
+  const [pictureId, setPictureId] = useState<string>(fallbackPictureId);
+  const [gameOver, setGameOver] = useState(false);
+  const [needsExplanation, setNeedsExplanation] = useState(false);
+  const { data } = useGetPictureInfoQuery({
+    variables: { pictureId: pictureId || fallbackPictureId },
+  });
+  const picture: FlatPicture | undefined = useSimplifiedQueryResponseData(data)?.picture;
+  const pictureLink = picture?.media?.url
+    ? asApiPath(`${picture.media.url}?updatedAt=${picture.media.updatedAt as string}`)
+    : '';
+
+  const onNextPicture = () => {
+    const nextPicture = getNextPicture();
+    nextPicture ? setPictureId(nextPicture) : setGameOver(true);
+  };
+  const { data: geoData } = useGetPictureGeoInfoQuery({ variables: { pictureId } });
```

Comment:
I think you don't need 90% of the stuff getPictureInfo gives you here, maybe combine both of these queries by writing a picture query that only gives you the picture media information and the geo information (as subfield of the picture query)?